resources/fetching/fetchers/azure/assets_enricher_storage_account.go

```diff
@@ -70,7 +76,54 @@ func (*storageAccountEnricher) addUsedForActivityLogsFlag(assets []inventory.Azu
 		if a.Extension == nil {
 			a.Extension = make(map[string]any)
 		}
-		a.Extension["usedForActivityLogs"] = true
+		a.Extension[inventory.ExtensionUsedForActivityLogs] = true
+		assets[i] = a
+	}
+}
+
+func (e storageAccountEnricher) addStorageAccountBlobServices(ctx context.Context, assets []inventory.AzureAsset) error {
+	storageAccounts := lo.Filter(assets, func(item inventory.AzureAsset, index int) bool {
+		return item.Type == inventory.StorageAccountAssetType
+	})
+	if len(storageAccounts) == 0 {
+		return nil
+	}
+
+	dataProtection, err := e.provider.ListStorageAccountBlobServices(ctx, storageAccounts)
+	if err != nil {
+		return err
+	}
+
+	if len(dataProtection) == 0 {
+		return nil
+	}
+
+	perStorageAccountID := map[string]inventory.AzureAsset{}
+	for _, d := range dataProtection {
+		perStorageAccountID[strings.FromMap(d.Extension, "storageAccountId")] = d
+	}
+
+	var errAgg error
+	for i, a := range assets {
+		if a.Type != inventory.StorageAccountAssetType {
+			continue
+		}
+
+		dataProtectionSettings, exist := perStorageAccountID[a.Id]
+		if !exist {
+			continue
+		}
+		if a.Extension == nil {
```

Comment:
Init it always might be challenging to ensure since we do not have any way to enforce a constructor for `AzureAsset.` 

We could embed this logic inside a getter. 

I am wondering how straightforward that API would be, though. (When do someone calls the getter when the Extension directly. Unles we make the extensions private?)

Reply:
Perhaps something like `AddExtension(string, any)` that adds the specific key to the extension map. Is that what you mean? 

I think this might be nice, I will do it and update the rest of the sets.

Reply:
Done.